shared/src/main/scala/coop/rchain/shared/MapOps.scala

```diff
@@ -0,0 +1,8 @@
+package coop.rchain.shared
+
+object MapOps {
+  def merge[K, V1, V2](map1: Map[K, V1], map2: Map[K, V2], v1: V1, v2: V2): Map[K, (V1, V2)] =
```

Comment:
The name merge is not right. See here what people understand by map merging https://www.google.com/search?q=map+datastructure+merge&oq=map+datastructure+merge&aqs=chrome..69i57j69i64.4675j0j4&sourceid=chrome&ie=UTF-8

Maybe something like `zip` would be better. Or do you have other ideas?